### PR TITLE
Fix UpdateOutput of logger by updating the correct outputs only.

### PR DESCRIPTION
**Pull request by shobhit85**
This change fixes the updating of outputs of a logger. Each output
has a set of classes for which it is supposed to be applied on a line
data. In this change an intersection of classes is taken between an
output's classes and given classes to be updated. If intersection is
same as output's set of classes then whole output is updated otherwise
output is split into two, one for intersection and other for the rest.
Intersection one is updated with new URI but old one still keeps old URI.

For example if an output has been defined for DEBUGPLUS (= DEBUG | INFO | WARN | ERROR | FATAL | isPLUSDEF) class then applying an update for ALL will update this output and NO new output will be added.

But on the other hand if an update is applied for INFO then a new output wrapper will be generated with new URI for INFO class and old output wrapper will be saved for (DEBUGPLUS ^ INFO) & DEBUGPLUS = (DEBUG | WARN | ERROR | FATAL | isPLUSDEF).

Further if an update comes for ALL class then both of the output wrappers will be updated (INFO & (DEBUG | WARN | ERROR | FATAL | isPLUSDEF)) and no new output wrapper will be generated.

@wallyqs @olegshaldybin @jpoler @vitroth

### CODE REVIEW

**Comment by wallyqs** [comment]
LGTM, thanks @shobhit85

File: logger_test.go [hunk 1]
```diff
@@ -27,12 +28,79 @@ func TestLoggerOutputs(t *testing.T) {
 
 	logger := New()
 
-	// Update for DEBUG should not update the output.
+	// Update for DEBUG should not 	update the output.
```
Comment (vitroth) [inline logger_test.go]: nit: extra space

**Comment by krobertson** [comment]
Should one of the test cases also verify an update to a different output scheme does not get updated?

**Comment by shobhit85** [comment]
@krobertson done!

**Comment by krobertson** [comment]
Thanks :smile:

**Pull request merged**